hathor/feature_activation/feature_service.py

```diff
@@ -47,8 +48,15 @@ def __init__(self, *, feature_settings: FeatureSettings, tx_storage: 'Transactio
         self._feature_settings = feature_settings
         self._tx_storage = tx_storage
 
-    def is_feature_active(self, *, block: 'Block', feature: Feature) -> bool:
-        """Returns whether a Feature is active at a certain block."""
+    def is_feature_active_for_transaction(self, *, tx: 'Transaction', feature: Feature) -> bool:
+        """Return whether a Feature is active for a certain Transaction."""
+        metadata = tx.get_metadata()
```

Comment:
Assert that `isinstance(tx, Transaction)` to prevent bugs if mypy fails to detect wrong call types.

Reply:
This has been removed.

Reply:
Ok. Thanks!